File: api/all/src/main/java/io/opentelemetry/api/common/KeyValue.java
/*
 * Copyright The OpenTelemetry Authors
 * SPDX-License-Identifier: Apache-2.0
 */

package io.opentelemetry.api.common;

/**
 * Key-value pair of {@link String} key and {@link Value} value.
 *
 * @see Value#of(KeyValue...)
 * @since 1.42.0
 */
public interface KeyValue {

  /** Returns a {@link KeyValue} for the given {@code key} and {@code value}. */
  static KeyValue of(String key, Value<?> value) {
    return KeyValueImpl.create(key, value);
  }

  /** Returns the key. */
  String getKey();

  /** Returns the value. */
  Value<?> getValue();
}

File: api/all/src/main/java/io/opentelemetry/api/common/KeyValue.kt
/*
 * Copyright The OpenTelemetry Authors
 * SPDX-License-Identifier: Apache-2.0
 */
package io.opentelemetry.api.common

/**
 * Key-value pair of [String] key and [Value] value.
 *
 * @see Value.of
 * @since 1.42.0
 */
interface KeyValue {
    /** Returns the key.  */
    val key: String?

    /** Returns the value.  */
    val value: Value<*>?

    companion object {
        /** Returns a [KeyValue] for the given `key` and `value`.  */
        fun of(key: String?, value: Value<*>?): KeyValue {
            return KeyValueImpl.create(key, value)
        }
    }
}
